### PR TITLE
TDD development

### PR DESCRIPTION
**Pull request by kimond**
@Pacane, I made this pull request to allow you to review my changes and discuss about them.

### CODE REVIEW

File: lib/src/metadata.dart [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Metadata {
+  Map<String, String> values = {};
```
Comment (Pacane) [inline lib/src/metadata.dart]: `final Map<String, String> values = <String, String>{}`

File: lib/src/parser.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    'discnumber',
+    'tracknumber',
+    'title',
+    'date',
+    'duration',
+    '!break!'
+  ];
+
+  Metadata parse(String cmusMetadata) {
+    var result = new Metadata();
```
Comment (Pacane) [inline lib/src/parser.dart]: `final` instead of `var` everywhere, where an object reference shouldn't change

File: lib/src/parser.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import 'package:cmustify/src/metadata.dart';
+
+class Parser {
+  final String breakKey = "!break!";
+  final List<String> validKeys = [
```
Comment (Pacane) [inline lib/src/parser.dart]: I'd call this `tag` instead of `key` maybe.

**Comment by Pacane** [comment]
other than that, LGTM

**Comment by kimond** [comment]
I took in consideration your modifications and I applied them.

**Pull request merged**